### DISCUSSION TITLE
auth.id() not working, returning null

### DISCUSSION THREAD

**Discussion by Jciel** [supabase/supabase#27193] [Questions] [answered]
Hi people,
Im using `auth.uid()` in a policy for slect operation for a table, but this not working, the function `auth.uid()` is retuning `null`
I check the jwt in request header and is correct, all that retuning 403 code to my frontend, Im using QuasarJS.
My policy is simple like that:
```
alter policy "Title policy"
as PERMISSIVE
for SELECT
on "public"."clients"
to anon, authenticated
using (
  ( (select auth.uid()) = personal_id )
);
```

**Comment by GaryAustin1** [comment]
How are you making  your select call?
How do you know auth.uid() is null?
A select policy will not error if you don't meet RLS.  It will return an empty array.
403 is usually a permission error (from no grant to the table or schema) to do an operation or access a schema or table.  There should be an error returned with the 403 with more details, or look in the API Gateway log.

**Comment by beaux-riel** [comment]
I could be wrong, but have you tried dropping 'select':

alter policy "Title policy"
as PERMISSIVE
for SELECT
on "public"."clients"
to anon, authenticated
using (
    auth.uid() = personal_id
);

**Reply by GaryAustin1** [reply]
The select is there for performance reasons and should be fine.

**Comment by Jciel** [comment]
I noticed that my requests are going without the complete JWT token data, it seems that my JS client does not keep the JWT token

**Reply by GaryAustin1** [reply]
You'll have to provide a lot more info.  Not clear what complete jwt token data means.  Maybe you are getting anon token which means you don't have a signed in user in the client making the call.

**Reply by Jciel** [reply]
ok, I think I understand, I have a functionality within the application to register new users, using the supabase.auth.signInWithOtp() method, after I call this method I notice that it clears my current session, is this normal behavior?
This functionality is within an 'admin' area, that is, a user is already logged in to register new users,

**Reply by GaryAustin1** [reply, marked as answer]
Yes it will replace any previous user in the same client.  You would need to have completely separate clients that don't share cookies or local storage.

**Reply by Jciel** [reply]
I understand, is there any known strategy for supabase to obtain the behavior I want? using his ecosystem or something.
thank you anyway.

**Reply by GaryAustin1** [reply]
If you are not using SSR then you can create two supabase clients.  One for the admin and use the other client to register new users.   It may also be possible with SSR but I don't use it.   There is also admin.createUser if your code is running server side.

**Reply by Jciel** [reply]
Got it, I'll try something, thanks @GaryAustin1
